flink-runtime/src/main/java/org/apache/flink/runtime/dispatcher/MiniDispatcher.java

```diff
@@ -109,7 +119,11 @@ public MiniDispatcher(
 
 		if (executionMode == ClusterEntrypoint.ExecutionMode.NORMAL) {
 			// terminate the MiniDispatcher once we served the first JobResult successfully
-			jobResultFuture.whenComplete((JobResult ignored, Throwable throwable) -> shutDown());
+			jobResultFuture.whenComplete((JobResult result, Throwable throwable) -> {
```

Comment:
`throwable` isn't used. If `jobResultFuture` cannot be completed exceptionally, `thenAccept` should be used.

Reply:
Will update